modules/imaging_browser/php/imaging_session_controlpanel.class.inc

```diff
@@ -105,18 +105,23 @@ class Imaging_Session_ControlPanel
             $params['PCandID'] = $timePoint->getCandID();
             $params['PVL']     = $timePoint->getVisitLabel();
         }
+
         // To enable DICOM download: Add pair: tarchiveID, ArchiveLocation
         $qresult = $DB->pselect(
-            "SELECT TarchiveID, ArchiveLocation 
-            FROM tarchive 
-            WHERE PatientName LIKE $ID",
+            "SELECT t.TarchiveID, t.ArchiveLocation, 
+                SUBSTRING_INDEX(mu.UploadLocation, '/', -1) AS originalFileName,
```

Comment:
I am not sure to understand why you are joining the mri_upload table. Also, doing that might break projects that are older than the mri_upload table (IBIS, PREVENT-AD...). For the renaming, I would simply append the patient name at the end of the archiveLocation. This way you have access to the patient name and can retraced from which archiveLocation this was downloaded if needed.

For example: `DCM_2017-04-10_ImagingUpload-11-33-4KNC3m.tar` could be renamed `DCM_2017-04-10_ImagingUpload-11-33-4KNC3m_MTL0123_456789_V01.tar`

Reply:
Also, the query should be prepared

Reply:
Also, if the `SessionID` has not been set by the imaging pipeline in the tarchive table, this query will return nothing and the download will download the DICOM archive with `ArchiveLocation` as the name. So `DCM_2017-04-10_ImagingUpload-11-33-4KNC3m.tar` would be downloaded as `DCM_2017-04-10_ImagingUpload-11-33-4KNC3m.tar`, no patient name info...

Reply:
The reason for joining the mri_upload table was so that we are able to retrieve the original filename and download the file with that name as that is what the users will be familiar with.
 It is not guaranteed that the original file name will be equivalent to the `PatientName` as you can add a suffix to the file that is being uploaded after the patient name (ex. `PatientName_anySuffix.zip`). 
If there are projects that do not have the mri_upload tables, then I agree we should just use the `ArchiveLocation` and append the PatientName. 

In which case will the `SessionID` not be set by the imaging pipeline? Do you suggest I remove the `t.sessionID=mu.sessionID` in the where clause?

Reply:
It is possible that the sessionID is null in the tarchive table if no MINC files were inserted into the files table. I guess for the imaging browser, it does not matter since you are looking for inserted MINC files in that module. Might be important to remember when doing the download from the DICOM archive module.